### ISSUE TITLE
SConscript(non-existing-file) is only a warning

### ISSUE THREAD

**Issue by mwichmann** [SCons/scons#3162]
Maling list discussion: see https://pairlist4.pair.net/pipermail/scons-users/2018-July/007111.html
TL;DR: calling SConscript() with a file that does not exist is a warning, not an error.
It is possible to argue that the current behavior is correct just as it is that it is wrong.  Life would be better if at least there is a way for a user to make sure they get hit with a big hammer if they want, if they leave something out, for example, when running a test suite.
Some approaches:

- adding a warning-control framework to the warning framework would allow some or all of the warnings to be classified as errors. 
    - Syntax in the style of gcc, `--Werror=`, is one possibility
    - Syntax like the existing scons `--warn=` (pretty similar) is another
    - the Python warnings module syntax provides another possible option (something like `
warnings.filterwarnings("error")` in code, with some syntax for commandline). This is more flexible than just warning=>error, it provides setting actions from the set (default, error, ignore, always, module, once), but would require reworking the scons warnings framework to match the Python one.
- modify the SConscript call to take a keyword argument, such as must_exist, which controls whether the call fails if the file is missing or not.  This does not affect any other warnings, and could have a suitable deprecation message and deprectation period with an option to switch to the new behavior immediately.  PR #3154 proposes this approach.
